sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -807,19 +811,39 @@ class ExecChecks private(
     tagUnsupportedTypes(meta, check, allowDecimal,
       meta.childPlans.flatMap(_.outputAttributes.map(toStructField)),
       "unsupported data types in input: %s")
+
+    extendedChecks.foreach {
+      case (fieldName, typeSig) =>
+        val fieldMeta = meta.namedChildExprs(fieldName)
+          .flatMap(_.typeMeta.dataType)
+          .zipWithIndex
+          .map(t => StructField(s"c${t._2}", t._1))
+        tagUnsupportedTypes(meta, typeSig, allowDecimal, fieldMeta,
+          s"unsupported data types in '$fieldName': %s")
+    }
```

Comment:
This is great. I just had another thought. It would be good as an internal sanity check that we also verify that there is a check for each key in `namedChildExprs`.  We already will throw an exception if `fieldName` is not in `namedChildExprs`. We should just make sure that both are tied to each other.

Reply:
I added a check for this and manually tested by introducing a typo in the key of one of the hash maps and got the error `GpuBroadcastHashJoinMeta is missing ExecChecks for rightKeys`.